qutebrowser/config/configdata.yml

```diff
@@ -2512,6 +2512,22 @@ colors.webpage.prefers_color_scheme_dark:
 
 # emacs: '
 
+## unkown_url
+
+unknown_url.scheme.policy:
+  default: 1
+  type:
+    name: Int
+    minval: 1
+    maxval: 3
+  desc: >-
+    Set UnknownUrlSchemePolicy.
+
+    Only valid for webengine backend (Qt > 5.10).
+    1: DisallowUnknownUrlSchemes, 2: AllowUnknownUrlSchemesFromUserInteraction, 3: AllowAllUnknownUrlSchemes.
```

Comment:
I think there are various things which should be changed here. In general, settings are user-facing and should hide implementation details from the underlying engine(s). As-is, this setting (and its values) is pretty user-unfriendly.

Some suggestions:

- Use a string with `valid_values` set (see other settings in the file for some examples), so that the options have a name rather than being a number you need to describe in the documentation. You'll then need to map that string to the enum member inside `set_unknown_url_scheme_policy`. See the [Qt documentation](https://doc.qt.io/qt-5/qwebenginesettings.html#UnknownUrlSchemePolicy-enum) for some inspiration about the descriptions for the values.
- Use the existing facilities to note compatibility instead of having it in the text, something like:

```
  backend:
    QtWebEngine: Qt 5.10
    QtWebKit: false
```

- Have a better text than "Set UnknownUrlSchemePolicy" - something that tells an user what this setting *does* exactly. Something like "How navigation requests to URLs with unknown schemes are handled." for example.

Reply:
Thanks for the all the handy suggestions! Updated in the latest commit!